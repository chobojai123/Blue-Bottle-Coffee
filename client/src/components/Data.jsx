import React, { Component } from 'react';
import { connect } from 'react-redux';
import moment from 'moment';

export class Data extends Component {
  constructor(props) {
    super(props);
    this.renderOrders.bind(this.renderOrders);
  }

  renderOrders() {
    const { orders, pageNumber } = this.props;

    const [start, end] = [
      (pageNumber - 1) * (20 + 5),
      pageNumber * 20 + pageNumber * 5,
    ];

    return orders !== null
      ? [...orders.slice(start, end)].map(order => (
          <tr key={order.coffee + order.orderNum}>
            <td>{order.coffee}</td>
            <td>{order.coffee}</td>
            <td>{order.cases}</td>
            <td>{order.packets}</td>
            <td>
              {order.priority
                ? `${moment(order.shipDate).format('l')}  *`
                : moment(order.shipDate).format('l')}
            </td>
            <td className="td-orderNum">{`#${order.orderNum}`}</td>
            <td className="td-view">
              <ion-icon name="eye" />
            </td>
          </tr>
        ))
      : null;
  }

  render() {
    return (
      <div className="table">
        <h5>Orders</h5>
        <hr />
        <table>
          <thead>
            <tr>
              <th className="th-coffee">Coffee</th>
              <th className="th-method">Method</th>
              <th className="th-cases">Number of Cases</th>
              <th className="th-packets">Packets per Case</th>
              <th className="th-shipDate">Ship Date</th>
              <th className="th-orderNum">Order</th>
              <th className="th-view">View</th>
            </tr>
          </thead>
          <tbody>{this.renderOrders()}</tbody>
        </table>
      </div>
    );
  }
}

function mapStateToProps({ orders }) {
  return { orders };
}

export default connect(mapStateToProps)(Data);
