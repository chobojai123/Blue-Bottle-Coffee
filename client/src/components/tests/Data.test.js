import React from 'react';
import { shallow, mount } from 'enzyme';
import '../../setupTests';
import { Data } from '../Data';

describe('Data component', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<Data orders={[1, 2, 3, 4, 5]} />);
  });

  test('renders', () => {
    expect(wrapper.exists()).toBe(true);
  });

  test('should have 7 headers', () => {
    expect(wrapper.find('th')).toHaveLength(7);
  });
});
