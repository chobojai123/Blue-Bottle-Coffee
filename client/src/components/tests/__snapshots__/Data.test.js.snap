// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Data component renders 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Data />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h5>
          Orders
        </h5>,
        <hr />,
        <table>
          <thead>
            <tr>
              <th
                className="th-coffee"
              >
                Coffee
              </th>
              <th
                className="th-method"
              >
                Method
              </th>
              <th
                className="th-cases"
              >
                Number of Cases
              </th>
              <th
                className="th-packets"
              >
                Packets per Case
              </th>
              <th
                className="th-shipDate"
              >
                Ship Date
              </th>
              <th
                className="th-orderNum"
              >
                Order
              </th>
              <th
                className="th-view"
              >
                View
              </th>
            </tr>
          </thead>
          <tbody />
        </table>,
      ],
      "className": "table",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Orders",
        },
        "ref": null,
        "rendered": "Orders",
        "type": "h5",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": "hr",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <thead>
              <tr>
                <th
                  className="th-coffee"
                >
                  Coffee
                </th>
                <th
                  className="th-method"
                >
                  Method
                </th>
                <th
                  className="th-cases"
                >
                  Number of Cases
                </th>
                <th
                  className="th-packets"
                >
                  Packets per Case
                </th>
                <th
                  className="th-shipDate"
                >
                  Ship Date
                </th>
                <th
                  className="th-orderNum"
                >
                  Order
                </th>
                <th
                  className="th-view"
                >
                  View
                </th>
              </tr>
            </thead>,
            <tbody />,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <tr>
                <th
                  className="th-coffee"
                >
                  Coffee
                </th>
                <th
                  className="th-method"
                >
                  Method
                </th>
                <th
                  className="th-cases"
                >
                  Number of Cases
                </th>
                <th
                  className="th-packets"
                >
                  Packets per Case
                </th>
                <th
                  className="th-shipDate"
                >
                  Ship Date
                </th>
                <th
                  className="th-orderNum"
                >
                  Order
                </th>
                <th
                  className="th-view"
                >
                  View
                </th>
              </tr>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <th
                    className="th-coffee"
                  >
                    Coffee
                  </th>,
                  <th
                    className="th-method"
                  >
                    Method
                  </th>,
                  <th
                    className="th-cases"
                  >
                    Number of Cases
                  </th>,
                  <th
                    className="th-packets"
                  >
                    Packets per Case
                  </th>,
                  <th
                    className="th-shipDate"
                  >
                    Ship Date
                  </th>,
                  <th
                    className="th-orderNum"
                  >
                    Order
                  </th>,
                  <th
                    className="th-view"
                  >
                    View
                  </th>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Coffee",
                    "className": "th-coffee",
                  },
                  "ref": null,
                  "rendered": "Coffee",
                  "type": "th",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Method",
                    "className": "th-method",
                  },
                  "ref": null,
                  "rendered": "Method",
                  "type": "th",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Number of Cases",
                    "className": "th-cases",
                  },
                  "ref": null,
                  "rendered": "Number of Cases",
                  "type": "th",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Packets per Case",
                    "className": "th-packets",
                  },
                  "ref": null,
                  "rendered": "Packets per Case",
                  "type": "th",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Ship Date",
                    "className": "th-shipDate",
                  },
                  "ref": null,
                  "rendered": "Ship Date",
                  "type": "th",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Order",
                    "className": "th-orderNum",
                  },
                  "ref": null,
                  "rendered": "Order",
                  "type": "th",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "View",
                    "className": "th-view",
                  },
                  "ref": null,
                  "rendered": "View",
                  "type": "th",
                },
              ],
              "type": "tr",
            },
            "type": "thead",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [],
            },
            "ref": null,
            "rendered": Array [],
            "type": "tbody",
          },
        ],
        "type": "table",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h5>
            Orders
          </h5>,
          <hr />,
          <table>
            <thead>
              <tr>
                <th
                  className="th-coffee"
                >
                  Coffee
                </th>
                <th
                  className="th-method"
                >
                  Method
                </th>
                <th
                  className="th-cases"
                >
                  Number of Cases
                </th>
                <th
                  className="th-packets"
                >
                  Packets per Case
                </th>
                <th
                  className="th-shipDate"
                >
                  Ship Date
                </th>
                <th
                  className="th-orderNum"
                >
                  Order
                </th>
                <th
                  className="th-view"
                >
                  View
                </th>
              </tr>
            </thead>
            <tbody />
          </table>,
        ],
        "className": "table",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Orders",
          },
          "ref": null,
          "rendered": "Orders",
          "type": "h5",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "hr",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <thead>
                <tr>
                  <th
                    className="th-coffee"
                  >
                    Coffee
                  </th>
                  <th
                    className="th-method"
                  >
                    Method
                  </th>
                  <th
                    className="th-cases"
                  >
                    Number of Cases
                  </th>
                  <th
                    className="th-packets"
                  >
                    Packets per Case
                  </th>
                  <th
                    className="th-shipDate"
                  >
                    Ship Date
                  </th>
                  <th
                    className="th-orderNum"
                  >
                    Order
                  </th>
                  <th
                    className="th-view"
                  >
                    View
                  </th>
                </tr>
              </thead>,
              <tbody />,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <tr>
                  <th
                    className="th-coffee"
                  >
                    Coffee
                  </th>
                  <th
                    className="th-method"
                  >
                    Method
                  </th>
                  <th
                    className="th-cases"
                  >
                    Number of Cases
                  </th>
                  <th
                    className="th-packets"
                  >
                    Packets per Case
                  </th>
                  <th
                    className="th-shipDate"
                  >
                    Ship Date
                  </th>
                  <th
                    className="th-orderNum"
                  >
                    Order
                  </th>
                  <th
                    className="th-view"
                  >
                    View
                  </th>
                </tr>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <th
                      className="th-coffee"
                    >
                      Coffee
                    </th>,
                    <th
                      className="th-method"
                    >
                      Method
                    </th>,
                    <th
                      className="th-cases"
                    >
                      Number of Cases
                    </th>,
                    <th
                      className="th-packets"
                    >
                      Packets per Case
                    </th>,
                    <th
                      className="th-shipDate"
                    >
                      Ship Date
                    </th>,
                    <th
                      className="th-orderNum"
                    >
                      Order
                    </th>,
                    <th
                      className="th-view"
                    >
                      View
                    </th>,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Coffee",
                      "className": "th-coffee",
                    },
                    "ref": null,
                    "rendered": "Coffee",
                    "type": "th",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Method",
                      "className": "th-method",
                    },
                    "ref": null,
                    "rendered": "Method",
                    "type": "th",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Number of Cases",
                      "className": "th-cases",
                    },
                    "ref": null,
                    "rendered": "Number of Cases",
                    "type": "th",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Packets per Case",
                      "className": "th-packets",
                    },
                    "ref": null,
                    "rendered": "Packets per Case",
                    "type": "th",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Ship Date",
                      "className": "th-shipDate",
                    },
                    "ref": null,
                    "rendered": "Ship Date",
                    "type": "th",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Order",
                      "className": "th-orderNum",
                    },
                    "ref": null,
                    "rendered": "Order",
                    "type": "th",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "View",
                      "className": "th-view",
                    },
                    "ref": null,
                    "rendered": "View",
                    "type": "th",
                  },
                ],
                "type": "tr",
              },
              "type": "thead",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [],
              },
              "ref": null,
              "rendered": Array [],
              "type": "tbody",
            },
          ],
          "type": "table",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
